include/dx2/reflection.hpp

```diff
@@ -706,36 +707,46 @@ class ReflectionTable {
   template <typename T>
   void add_column(const std::string &name, const std::vector<size_t> &shape,
                   const std::vector<T> &column_data) {
-    auto col = std::make_unique<TypedColumn<T>>(name, shape, column_data);
+    // Check if the type T is a bool. If so, convert to BoolEnum and add column.
+    if constexpr (std::is_same_v<T, bool>) {
+      std::vector<h5dispatch::BoolEnum> converted(column_data.size());
+      for (size_t i = 0; i < column_data.size(); ++i) {
+        converted[i] = column_data[i] ? h5dispatch::BoolEnum::TRUE
+                                      : h5dispatch::BoolEnum::FALSE;
+      }
+      add_column<h5dispatch::BoolEnum>(name, shape, converted);
+    } else {
+      auto col = std::make_unique<TypedColumn<T>>(name, shape, column_data);
 
-    // Check for duplicate column names
-    for (const auto &existing_col : data) {
-      if (existing_col->get_name() == name) {
-        throw std::runtime_error("Column with name already exists: " + name);
+      // Check for duplicate column names
+      for (const auto &existing_col : data) {
```

Comment:
The check for duplicate column names now needs to come before `// Check if the type T is a bool`, otherwise the check will only be made for non-bool types?